api/src/main/java/org/pmiops/workbench/reporting/insertion/InsertAllRequestPayloadTransformer.java

```diff
@@ -70,11 +68,7 @@ default RowToInsert modelToRowToInsert(MODEL_T model, Map<String, Object> fixedV
     final ImmutableMap.Builder<String, Object> columnToValueBuilder = ImmutableMap.builder();
     columnToValueBuilder.putAll(fixedValues); // assumed to have non-null values
     Arrays.stream(getQueryParameterColumns())
-        .map(
-            col ->
-                new AbstractMap.SimpleImmutableEntry<>(
-                    col.getParameterName(), col.getRowToInsertValue(model)))
-        .filter(e -> Objects.nonNull(e.getValue()))
+        .flatMap(col -> col.getRowToInsertEntry(model))
```

Comment:
flatMap() makes it easy to combine with null-filtering, which is a thing we need here